_posts/2021-02-08-ill-be-back.md

```diff
@@ -0,0 +1,60 @@
+
+Acme Inc. issues invoices on the last day of the month. All issued invoices are due in 30 days. It seems reasonable to set the "check overdue invoices" clock to tick on the 10 of the next month. Forty days after sending invoices, Acme checks payments and act upon overdue invoices. One day Acme changed its business model and decided to issue invoices on the same day they sell goods or services. The Acme accounting department cannot anymore rely on the Forty-day rule to check for overdue invoices because by issuing invoices not regularly, invoices can be outstanding more or less every day. The Acme Operations team's simple reaction is to adjust the ticking clock's scale, and instead of ticking on the 10 of the next month, it makes it tick every day.
+
+The overall system performances are greatly affected. Every 24 hours, there is a job that scans the database looking for overdue invoices. Given that Acme doesn't sell goods or services every day, it doesn't issue invoices either. The check for unpaid invoices job ends up overloading the system daily for no good reason.
+
+## Design problems at play
+
+The problem we're facing is not a technical problem. Sure, we can scale up the database server and throw some money at the problem. However, it's only postponing the issue to a later time; it's like hiding the dust under the carpet. I remember talking with a customer where the infamous batch job ended up being so complicated that it could only run once every week. The job was taking up to 3 days to complete, and the 4-day buffer was required to guarantee a viable human intervention window in case something terrible happened.
+
+Again, it's not a technical problem. It's a design issue.
+
+We're replicating the user mental model. Users always worked this way, they sit down and check for unpaid invoices. It doesn't matter if they do it every forty days or daily; this is how most accounting departments worked for their entire existence.
+
+### Flip the responsibility 
+
+If, 20 years ago, we were to ask someone: how do you know what your duties are for today? The most common answer would have been: I'll check my agenda every morning.
+
+The same question today gets an entirely different answer: I'll rely on my phone to notify that something is due, for example, in the next hour.
+
+We stopped relying on us looking at agendas, and instead, we rely on the agenda taking care of us. We mostly look at the schedule to get an overall view of the day if we want.
+
+What if we were to schedule in our calendar each time an invoice is issued that, forty days later, we have to check for the payment of that specific invoice? At this point, there is no need for a ticking clock to tell us regularly to check for overdue invoices; we'll rely on the agenda to come back to us with a notification that today there are invoices that we need to check.
```

Comment:
This seems to be setting the reader up for something and then it ends kind of abruptly. I would move the `In a future article` part at the end here so it flows better: What if our agendas could notify us? In a future article I'll show you how that works. Followed by the conclusion